runtime/src/main/java/org/corfudb/runtime/object/StreamId.java

```diff
@@ -0,0 +1,56 @@
+package org.corfudb.runtime.object;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.util.Utils;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@EqualsAndHashCode
+@ToString
+public class StreamId {
+    @Getter
+    private final UUID id;
+
+    private final Optional<String> name;
+
+    /**
+     * @param id id of the stream
+     */
+    private StreamId(UUID id, Optional<String> name) {
+        this.name = name;
+        this.id = id;
+    }
+
+    /**
+     * Either of the variables should be defined to create a streamId.
+     *
+     * @param streamName name of a stream
+     * @param uuid       id of a stream
+     * @return streamId object
+     * @throws IllegalStateException exception if none are defined
+     */
+    public static StreamId build(Optional<String> streamName, Optional<UUID> uuid) {
+        StreamId streamId;
```

Comment:
You can use a builder here.
Also, I think that overloading is more appropriate than passing optional here, since a stream can be identified in two ways. Either by name, or by UUID.